crates/oracle/src/models.rs

```diff
@@ -59,6 +59,7 @@ impl FromStr for Caip2ChainId {
         let is_ascii_alphanumberic_or_hyphen =
             |s: &str| s.chars().all(|c| c.is_ascii_alphanumeric() || c == '-');
 
+        #[allow(clippy::len_zero)]
```

Comment:
I decided to leave this part untouched because it is more legible this way than as suggested by the clippy lint.
This is subjective, so feel free to apply the lint instead.